### ISSUE TITLE
Fails to run with message "unexpected token `deleted'"

### ISSUE THREAD

**Issue by ElvenSpellmaker** [adsr/phpspy#77]
```
./phpspy -V72 -p 127743
sh: -c: line 0: syntax error near unexpected token `deleted'
sh: -c: line 0: `objdump -p /proc/127743/root//usr/bin/php;5cde72be (deleted) | awk '/LOAD/{print $5; exit}''
popen_read_line: No stdout; cmd=objdump -p /proc/127743/root//usr/bin/php;5cde72be (deleted) | awk '/LOAD/{print $5; exit}'
get_php_base_addr: Failed to get virt_addr
```

For some reason the binary seems to be having `;5cde72be (deleted)` appended which causes this to fail.

Running the command without this suffix gives me an address back.
```
objdump -p /proc/127743/root//usr/bin/php | awk '/LOAD/{print $5; exit}'
0x0000000000000000
```

This is on CentOS using Webtatic php72w.

**Comment by adsr** [comment]
Thanks for the report @ElvenSpellmaker. I could not repro with a vanilla Cent7 VM and php72w. What kind of PHP process are you tracing? (Apache worker / mod_php, php-fpm, php cli process?) Looks like a cli process running in a container but want to clarify.

`(deleted)` usually indicates the file was unlinked but some process has the inode open still. Is there anything in your setup that might be replacing the binary?

Also can you share the output of these commands? These are the commands executed in `get_php_bin_path`.

```shell
$ awk '/libphp7/{print $NF; exit 0} END{exit 1}' /proc/<pid>/maps
$ readlink /proc/<pid>/exe
```

**Comment by ElvenSpellmaker** [comment]
Thanks for your reply, it's a long running cli process (frontend of an IRC bot) which I wanted to debug further. It's running on a VM in Azure.

I reckon you're right, I reckon I've upgraded the php version to the latest 7.2 while the binary is still in use (and didn't restart it) and so it can no longer find the old one needed to run this. I should've recognised this quicker haha.

Unfortunately I've since restarted the process (so it works again as it was needed) and so I can't run the above.

I can now attach to the new process, confirming what you said above.
